### ISSUE TITLE
Allow users to specify which syntax a file renders with in Sourcegraph UI

### ISSUE THREAD

**Issue by christinelovett** [sourcegraph/sourcegraph#40176]
#### Feature request description

While viewing a file, allow a user to specify which syntax they want the file to render in. (Rather than having to explicitly set up in a Sourcegraph configuration location)

#### Is your feature request related to a problem? If so, please describe.

A user was viewing a shell script in a config file that isn't standardly named, and they want to see the shell script syntax highlighting instead of raw text. 

#### Describe alternatives you've considered.

Manually configure what language the file should be highlighted as, based on the file extension pattern, in Site Config. 

#### Additional context
The files in question are not scripts, they are config files, so it would be misleading to introduce a shebang at the top of them.
The user came across [sshrc config files](https://www.oreilly.com/library/view/ssh-the-secure/0596008953/ch08s04.html) named sshrc and rc. This may also apply to other rc files which tend to use sh syntax.

**Comment by olafurpg** [comment]
@christinelovett Is this a feature request from a customer?

**Comment by christinelovett** [comment]
@olafurpg yes, here is the [product gap](https://github.com/sourcegraph/accounts/issues/578) and [customer](https://github.com/sourcegraph/accounts/issues/578).

**Comment by olafurpg** [comment]
@christinelovett can you update the "product gap" link? It currently points to the same URL as the "customer" link

**Comment by christinelovett** [comment]
Whoops :-) fixed ✔️

**Comment by albertocavalcante** [comment]
What's the current status of this issue?
